### ISSUE TITLE
Housekeeping doesn't run during a long sleep

### ISSUE THREAD

**Issue by reidsunderland** [MetPX/sarracenia#968]
This was discovered in a poll with `sleep 1h`. The housekeeping interval is the default `300s` and sanity_log_dead is also the default, `450s`.

When logLevel is the default (INFO), no messages are output to the log for 1 hour which causes sanity to restart the poll after 7.5 minutes, thinking it's hung.

But housekeeping should be running every 5 minutes and outputting a log message.

```
2024-03-11 18:55:03,664 [DEBUG] sarracenia.flow run  not throttling: limit: 0
2024-03-11 18:55:03,664 [DEBUG] sarracenia.flow run sleeping for more than 60 seconds: 3466.35 seconds. Elapsed since wakeup: 133.65 Sleep setting: 3600.00
2024-03-11 18:55:03,664 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:08,666 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:13,672 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:18,677 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:23,682 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:28,688 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:33,693 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:38,698 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:43,702 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:48,706 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:53,712 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:55:58,717 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:03,722 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:08,728 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:13,733 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:18,738 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:23,744 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:28,749 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:33,754 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:38,760 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:43,765 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:48,770 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:53,776 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:56:58,781 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:03,786 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:08,788 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:13,793 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:18,799 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:23,804 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:28,809 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:33,815 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:38,820 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:43,825 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:48,831 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:53,836 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:57:58,841 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:03,847 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:08,852 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:13,857 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:18,858 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:23,864 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:28,869 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:33,874 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:38,880 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:43,885 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:48,890 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:53,896 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:58:58,901 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:03,906 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:08,912 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:13,917 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:18,922 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:23,928 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:28,933 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:33,938 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:38,944 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:43,949 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:48,954 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:53,960 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 18:59:58,965 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:03,970 [DEBUG] sarracenia.flow run sleeping for 5.00
# Housekeeping should have ran somewhere near here and output an INFO level log
2024-03-11 19:00:08,974 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:13,980 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:18,985 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:23,990 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:28,994 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:33,998 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:39,002 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:44,008 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:49,013 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:54,018 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:00:59,024 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:04,029 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:09,034 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:14,040 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:19,045 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:24,050 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:29,056 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:34,061 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:39,066 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:44,070 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:49,074 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:54,078 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:01:59,082 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:04,088 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:09,093 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:14,098 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:19,104 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:24,106 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:29,112 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:34,117 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:39,122 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:44,128 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:49,133 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:54,138 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:02:59,142 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:04,147 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:09,153 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:14,158 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:19,163 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:24,169 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:29,174 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:34,179 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:39,185 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:44,190 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:49,195 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:54,201 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:03:59,206 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:04,211 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:09,217 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:14,222 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:19,228 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:24,233 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:29,238 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:34,244 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:39,249 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:44,254 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:49,260 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:54,265 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:04:59,270 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:04,274 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:09,280 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:14,282 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:19,288 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:24,290 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:29,294 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:34,298 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:39,302 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:44,308 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:49,310 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:54,314 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:05:59,318 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:04,324 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:09,329 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:14,334 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:19,340 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:24,345 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:29,350 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:34,356 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:39,361 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:44,366 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:49,372 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:54,377 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:06:59,382 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:04,386 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:09,390 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:14,396 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:19,401 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:24,406 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:29,411 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:34,417 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:39,418 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:44,423 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:49,429 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:54,430 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:07:59,436 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:04,438 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:09,442 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:14,448 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:19,453 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:24,458 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:29,462 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:34,468 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:39,473 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:44,478 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:49,482 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:54,488 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:08:59,493 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:04,498 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:09,504 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:14,509 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:19,514 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:24,518 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:29,524 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:34,526 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:39,532 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:44,537 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:49,542 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:54,548 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:09:59,553 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:04,554 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:09,558 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:14,564 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:19,569 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:24,574 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:29,578 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:34,582 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:39,586 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:44,590 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:49,594 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:54,600 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:10:59,602 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:04,606 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:09,611 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:14,614 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:19,618 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:24,624 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:29,625 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:34,631 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:39,636 [DEBUG] sarracenia.flow run sleeping for 5.00
2024-03-11 19:11:44,638 [DEBUG] sarracenia.flow run sleeping for 5.00
```

**Comment by petersilva** [comment]
Can we talk about the long periods where messages are queued for polls in this issue,
or should it be a separate issue.  I'm trying to understand how big the impact is of the queueing.

**Comment by reidsunderland** [comment]
I just created a separate issue #974

**Comment by reidsunderland** [comment]
I think this can be closed, because it's separate. 

Re-working polls to act like scheduled flows will also cause housekeeping to run more often, because the sleep time will be shorter.

The change I implemented in #969 will ensure housekeeping is run during any long sleep in the flow algorithm, regardless of the component.
